src/nile/core/account.py

```diff
@@ -172,15 +188,19 @@ def send(
             query_flag=query_type,
         )
 
-    def simulate(self, address_or_alias, method, calldata, max_fee=None, nonce=None):
+    async def simulate(
+        self, address_or_alias, method, calldata, max_fee=None, nonce=None
+    ):
         """Simulate a tx going through an Account contract."""
-        return self.send(address_or_alias, method, calldata, max_fee, nonce, "simulate")
+        return await self.send(
```

Comment:
what happens if we don't `await` in here? does it work the same? i.e:
- waiting for `send` if `await simulate()`
- not waiting for anything otherwise

Reply:
Good question! `simulate` and `estimate_fee` don't need to be async

Reply:
they do need to be async. just this line i don't think it needs to be awaited, the user of `simulate()` can `await` when calling it with the same result, no?